packages/jest-util/src/setGlobal.ts

```diff
@@ -6,12 +6,16 @@
  */
 
 import type {Global} from '@jest/types';
+import {isShreddable, setNotShreddable} from './shredder';
 
 export default function setGlobal(
   globalToMutate: typeof globalThis | Global.Global,
-  key: string,
+  key: string | symbol,
   value: unknown,
+  shredAfterTeardown = true,
```

Comment:
Let's not do default true boolean values that require passing "false" to disable. That's a confusing API.

I recommend using a type like `'retain' | 'clear'` or similar, and require it being passed. That way developers can't make mistakes.